### PR TITLE
Cleanup and update for React 0.14.

### PR DESCRIPTION
**Pull request by dantman**
- Exclusively depend on ^0.14.0 of react
- Use react-addons-pure-render-mixin instead of an internal require that'll likely be deprecated
- In example and README examples:
  - Use render from react-dom instead of the deprecated one in react
  - Don't render directly into document.body, this causes a React warning

This should be a new breaking release.

### CODE REVIEW

**Comment by nkbt** [comment]
Thank you very much for your contribution! I would be happy to merge it in (everything seems totally OK for me in terms of code and reasoning), but I have some reasons not to:

I prefer to support both 0.13 and 0.14 at the same time. It is possible and compatible. Also there is a react-component-template and all my libs `extend` it. So any code that effects deps or other generic code should go there.

I gradually update other libs and react-interval will be updated very soon too.

Pure render is not deprecated in 0.14 so I don't see any reason to add extra dependency for now. When it is - sure thing.

File: src/example/Example.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import App from './App';
 
+const root = document.createElement('div');
```
Comment (nkbt) [inline src/example/Example.js]: This is a good idea feel free to PR this to https://github.com/nkbt/react-component-template

Comment (nkbt) [inline src/example/Example.js]: I'd also give it some ID like `#app` or something.

File: circle.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ machine:
 dependencies:
   pre:
     - npm update -g npm
-    - npm install react@0.13
+    - npm install react@^0.14
```
Comment (nkbt) [inline circle.yml]: This is already solved in https://github.com/nkbt/react-component-template

**Comment by dantman** [comment]
How do you intend to support both 0.13 and 0.14 simultaneously in all your components? At least not without deprecation warnings.

`React.render` and `React.findDOMNode` are deprecated in 0.14, so if you use them your components will emit deprecation warnings.

`react-dom` does not exist in 0.13. So you can't use `ReactDOM.render` and `React.findDOMNode` in 0.13.

ReactInterval doesn't use either in the component itself but your examples and other components that do use them will emit warnings.

---

`PureRenderMixin` within `react` **is** deprecated.

The reason `react-interval` does not get a warning is because it is explicitly requiring the `react/lib/ReactComponentWithPureRenderMixin` which is a React internal module **not** a public interface.

The documented public interface for 0.13 was `require('react/addons').addons.PureRenderMixin`. This does emit a warning because `react/addons` is deprecated.

The [docs](https://facebook.github.io/react/docs/pure-render-mixin.html) have been [updated](https://github.com/facebook/react/commit/c2ae33271860cfabcfa9daab109b17f9d1e34e1a#diff-d3c52804bb9bffb6ec0045d29dee9ebf) to use `require('react-addons-pure-render-mixin')`.

Basically instead of getting a new warning when 0.15 comes out, `react-interval` is just going to straight up break because it's likely that when the `react/addons` deprecation is removed `react/lib/ReactComponentWithPureRenderMixin` will be removed from `react` and moved explicitly into `react-addons-pure-render-mixin`.

**Comment by nkbt** [comment]
Thanks a lot  for an explanation. I double-checked couple of other components and they do give warnings. Did not see them since my build was done in prod mode.

That's all totally makes sense.

All these changes should go to https://github.com/nkbt/react-component-template though, so I'll have them available in all other libs as well. Do you want to PR them there?

**Comment by nkbt** [comment]
Hi, @dantman, I realized that incompatibility with react@0.13/0.14 is only related to the lib's example, not the lib itself. So I want to keep 0.13/0.14 as peer dep, but 0.14 as dev dep. In this way all examples can be built with react 0.14, but lib itself can work with both.

I've got couple of libs that might not be that compatible. I guess at least once I used ref somewhere.

Ah, and of course `react-addons-pure-render-mixin` should be used, so it is compatible.

I want 0.13 support because most of these libs are used in production and react update for those projects will be done a little bit later. But I do not really want to support two different versions for each lib for now.

**Comment by nkbt** [comment]
Could you please rebase changes? Though it actually feels like everything works without warnings. Even with 

```
import {shouldComponentUpdate} from 'react/lib/ReactComponentWithPureRenderMixin';
```

I tried react 13 and 14, both work fine (the only difference - for 13 I replaced ReactDOM with React to test). But when this is used as lib, then all is good.

File: README.md [hunk 3]
```diff
@@ -66,7 +67,7 @@ const App = React.createClass({
 });
 
 
-React.render(<App />, document.body);
+ReactDOM.render(<App />, appRoot);
```
Comment (nkbt) [inline README.md]: `appRoot` is undefined here in example

File: README.md [hunk 4]
```diff
@@ -112,7 +114,7 @@ const App = React.createClass({
...
```
Comment (nkbt) [inline README.md]: here too

**Comment by nkbt** [comment]
Rest is good, thanks

File: README.md (hunk 3 above)
Comment (dantman) [inline README.md]: That's intentional. These are README examples, not runnable examples. Or tutorials for React basics.

This variable just signifies rendering of the app into a dom node intended for the app, other than the body, which the developer should already have.

Comment (nkbt) [inline README.md]: I prefer to have examples that you can copy/paste and they will Just Work (tm)

Comment (dantman) [inline README.md]: It's not going to 'just work' without the html as well.

For that matter, the render isn't actually key to the example. It could simply be omitted like the html is.

Comment (nkbt) [inline README.md]: That makes sense. I guess since we are discouraged to render into body itself (which we've never done in real projects anyway), there is a fair bit of boilerplate code. Not good for examples at all. So better to remove this part.

**Comment by nkbt** [comment]
Awesome!

**Comment by dantman** [comment]
Thanks.

It would be nice to have a release with this so react-interval can be installed in projects using 0.14.

**Comment by nkbt** [comment]
Will do it today bit later

**Comment by nkbt** [comment]
Published 1.2.0
